ipn/ipnlocal/local.go

```diff
@@ -4791,3 +4798,23 @@ func (b *LocalBackend) ResetAuth() error {
 	}
 	return b.resetForProfileChangeLockedOnEntry()
 }
+
+// StreamDebugCapture writes a pcap stream of packets traversing
+// tailscaled to the provided response writer.
+func (b *LocalBackend) StreamDebugCapture(w http.ResponseWriter) error {
+	var s *capture.Sink
+
+	b.mu.Lock()
+	if b.debugSink == nil {
+		s = capture.New()
```

Comment:
why doesn't capture.New take the writer? what's up with the RegisterOutput step being separate?

Reply:
Because a sink can have multiple outputs. Its the sink's responsibility to fan out from the `LogPacket` call to each output.